src/D2L.Bmx/Browser.cs

```diff
@@ -0,0 +1,64 @@
+using PuppeteerSharp;
+
+namespace D2L.Bmx;
+
+public class Browser {
+
+	// https://github.com/microsoft/playwright/blob/6763d5ab6bd20f1f0fc879537855a26c7644a496/packages/playwright-core/src/server/registry/index.ts#L630
+	private static readonly string[] WindowsEnvironmentVariables = [
+		"LOCALAPPDATA",
+		"PROGRAMFILES",
+		"PROGRAMFILES(X86)",
+	];
+
+	// https://github.com/microsoft/playwright/blob/6763d5ab6bd20f1f0fc879537855a26c7644a496/packages/playwright-core/src/server/registry/index.ts#L457-L459
+	private static readonly string[] WindowsPartialPaths = [
+		"\\Google\\Chrome\\Application\\chrome.exe",
+		"\\Microsoft\\Edge\\Application\\msedge.exe"
+	];
+	private static readonly string[] MacPaths = [
+		"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
+		"/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge"
+	];
+	private static readonly string[] LinuxPaths = [
+		"/opt/google/chrome/chrome",
+		"/opt/microsoft/msedge/msedge"
+	];
+
+	public static async Task<IBrowser?> LaunchBrowserAsync( bool noSandbox = false ) {
+		string? browserPath = GetPathToBrowser();
+		if( browserPath is null ) {
+			return null;
+		}
+
+		var launchOptions = new LaunchOptions {
+			Headless = Environment.GetEnvironmentVariable( "BMX_DEBUG" ) != "1",
```

Comment:
So when you run it with ``BMX_DEBUG = 1`` wouldn't the chromium instance automatically get stopped once puppeteer is done with getting sessions.

Any situations when it would remain open long enough to do any real investigation?

Reply:
> wouldn't the chromium instance automatically get stopped once puppeteer

Yea it would close after it gets the session or the 15 second timeout

> Any situations when it would remain open long enough to do any real investigation?

Yea I'm not sure how valuable it could be. Knowing where it got stuck in the login process if it was the login screen or some other page could maybe give some insight

Reply:
Just the fact that currently there are no exceptions being thrown I feel like it might be hard to know what went wrong. I could check the env variable again in that generic exception catch though and also print the exception